frankenstein/querier.go

```diff
@@ -14,132 +14,33 @@
-		Timeout: timeout,
-	}
-	return &IngesterQuerier{
-		url:    url,
-		client: client,
-	}
-}
-
-// Query implements Querier.
-func (q *IngesterQuerier) Query(from, to model.Time, matchers ...*metric.LabelMatcher) (model.Matrix, error) {
-	req := &generic.GenericReadRequest{
-		StartTimestampMs: proto.Int64(int64(from)),
-		EndTimestampMs:   proto.Int64(int64(to)),
-	}
-	for _, matcher := range matchers {
-		var mType generic.MatchType
-		switch matcher.Type {
-		case metric.Equal:
-			mType = generic.MatchType_EQUAL
-		case metric.NotEqual:
-			mType = generic.MatchType_NOT_EQUAL
-		case metric.RegexMatch:
-			mType = generic.MatchType_REGEX_MATCH
-		case metric.RegexNoMatch:
-			mType = generic.MatchType_REGEX_NO_MATCH
-		default:
-			panic("invalid matcher type")
-		}
-		req.Matchers = append(req.Matchers, &generic.LabelMatcher{
-			Type:  &mType,
-			Name:  proto.String(string(matcher.Name)),
-			Value: proto.String(string(matcher.Value)),
-		})
-	}
-
-	data, err := proto.Marshal(req)
-	if err != nil {
-		return nil, err
-	}
-	buf := bytes.NewBuffer(data)
-
-	// TODO: This isn't actually the correct Content-type.
-	resp, err := q.client.Post(q.url, string(expfmt.FmtProtoDelim), buf)
-	if err != nil {
-		return nil, err
-	}
-	defer resp.Body.Close()
-	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("server returned HTTP status %s", resp.Status)
-	}
-
-	r := &generic.GenericReadResponse{}
-	buf.Reset()
-	_, err = buf.ReadFrom(resp.Body)
-	if err != nil {
-		return nil, fmt.Errorf("unable to read response body: %s", err)
-	}
-	err = proto.Unmarshal(buf.Bytes(), r)
-	if err != nil {
-		return nil, fmt.Errorf("unable to unmarshal response body: %s", err)
-	}
-
-	m := make(model.Matrix, 0, len(r.Timeseries))
-	for _, ts := range r.Timeseries {
-		var ss model.SampleStream
-		ss.Metric = model.Metric{}
-		if ts.Name != nil {
-			ss.Metric[model.MetricNameLabel] = model.LabelValue(ts.GetName())
-		}
-		for _, l := range ts.Labels {
-			ss.Metric[model.LabelName(l.GetName())] = model.LabelValue(l.GetValue())
-		}
-
-		ss.Values = make([]model.SamplePair, 0, len(ts.Samples))
-		for _, s := range ts.Samples {
-			ss.Values = append(ss.Values, model.SamplePair{
-				Value:     model.SampleValue(s.GetValue()),
-				Timestamp: model.Time(s.GetTimestampMs()),
-			})
-		}
-		m = append(m, &ss)
-	}
-
-	return m, nil
+	Query(ctx context.Context, from, to model.Time, matchers ...*metric.LabelMatcher) (model.Matrix, error)
 }
 
 // A ChunkQuerier is a Querier that fetches samples from a ChunkStore.
 type ChunkQuerier struct {
+	ctx   context.Context // TODO initialise this to something reasonable.
 	Store ChunkStore
 }
 
 // Query implements Querier and transforms a list of chunks into sample
 // matrices.
-func (q *ChunkQuerier) Query(from, to model.Time, matchers ...*metric.LabelMatcher) (model.Matrix, error) {
+func (q *ChunkQuerier) Query(_ context.Context, from, to model.Time, matchers ...*metric.LabelMatcher) (model.Matrix, error) {
```

Comment:
Hmm, why store a ctx in ChunkQuerier and then making this one here that's passed in unused?

My plan is that the MergeQuerier (which will be constructed for every request) will contain a context and then pass that in here...